### PR TITLE
fixing `fetchMyTrades` / `parseTrade` on bibox

### PR DESCRIPTION
**Pull request by benediamond**
again, #3073

I think this should do it?! @Showgofar please verify this to the extent you can. thank you.

### CODE REVIEW

**Comment by benediamond** [comment]
On my end, we have:
```python
>>> bibox.fetch_closed_orders("TRX/ETH")
[{'info':
    {'id': 620945165,
     'createdAt': 1528559796000,
     'account_type': 0,
     'coin_symbol': 'TRX',
     'currency_symbol': 'ETH',
     'order_side': 1,
     'order_type': 2,
     'price': '0.00009558',
     'amount': '110.0000',
     'money': '0.01051380',
     'deal_price': '0.00009558',
     'deal_amount': '110.0000',
     'deal_money': '0.01051380',
     'status': 3, 'order_from': 6},
  'id': '620945165',
  'timestamp': 1528559796000,
  'datetime': '2018-06-09T15:56:36.000Z',
  'lastTradeTimestamp': None,
  'symbol': 'TRX/ETH',
  'type': 'limit',
  'side': 'buy',
  'price': 9.558e-05,
  'amount': 110.0,
  'cost': 0.0105138,
  'filled': 110.0,
  'remaining': 0.0,
  'status': 'closed',
  'fee': None
 }
...
]
```
and
```python
>>> bibox.fetch_my_trades("TRX/ETH")
[{'id': '676586428',
  'info':
    {'id': 676586428,
     'createdAt': 1528559796000,
     'account_type': 0,
     'coin_symbol':
     'TRX', 'currency_symbol':
     'ETH', 'order_side': 1,
     'order_type': 2,
     'price': '0.00009558',
     'amount': '110.0000',
     'money': '0.01051380',
     'fee': '0.11000000',
     'pay_bix': 0,
     'fee_symbol': 'TRX'},
  'timestamp': 1528559796000,
  'datetime': '2018-06-09T15:56:36.000Z',
  'symbol': 'TRX/ETH',
  'type': 'limit',
  'side': 'buy', 'price': 9.558e-05,
  'amount': 110.0,
  'fee': {'cost': 0.11, 'currency': None}
 }
...
]
```
let me know if this looks right.

**Comment by Showgofar** [comment]
yes its looks more real
but as you can see there is no relations between trades and order (trades doesnt have initial order ID)

**Comment by benediamond** [comment]
yes, ok, but at least these correctly perform their expected functions. @kroitor let me know your thoughts

File: js/bibox.js [hunk 1]
```diff
@@ -220,17 +220,15 @@ module.exports = class bibox extends Exchange {
     }
 
     parseTrade (trade, market = undefined) {
-        let timestamp = trade['time'];
-        let side = this.safeInteger (trade, 'side');
-        side = this.safeInteger (trade, 'order_side', side);
+        let timestamp = trade['createdAt'];
```
Comment (kroitor) [inline js/bibox.js]: You did break the parsing for fetchTrades with this edit. Can you make parseTrade parse both the public and private trades, like it does for other exchanges? ) Thx!

Comment (kroitor) [reply js/bibox.js]: A public trade looks like this:

```
              {     id:  35131874,
                 pair: "ETH_BTC",
                price: "0.07678341",
               amount: "0.0873",
                 time:  1528748288000,
                 side:  2              },
```

Comment (benediamond) [reply js/bibox.js]: Thanks for the input. Please take a look at the latest

**Comment by benediamond** [comment]
thanks for your work.

**Comment by Showgofar** [comment]
Thank you - fetch_my_trades return trades again.
   
here is order example
 {
        "info": {
            "id": 630991073,
            "createdAt": 1528753861000,
            "account_type": 0,
            "coin_symbol": "ETH",
            "currency_symbol": "BTC",
            "order_side": 2,
            "order_type": 2,
            "price": "0.07677000",
            "amount": "0.1020",
            "money": "0.00783054",
            "deal_price": "0.07681211",
            "deal_amount": "0.1020",
            "deal_money": "0.00783483",
            "status": 3,
            "order_from": 6
        },
        "id": "630991073",
        "timestamp": 1528753861000,
        "datetime": "2018-06-11T17:51:01-04:00",
        "lastTradeTimestamp": null,
        "symbol": "ETH\/BTC",
        "type": "limit",
        "side": "sell",
        "price": 0.07677,
        "amount": 0.102,
        "cost": 0.00783054,
        "filled": 0.102,
        "remaining": 0,
        "status": "closed",
        "fee": null
    },

and here is corresponding trade

    {
        "info": {
            "id": 686384927,
            "createdAt": 1528753861000,
            "account_type": 0,
            "coin_symbol": "ETH",
            "currency_symbol": "BTC",
            "order_side": 2,
            "order_type": 2,
            "price": "0.07681212",
            "amount": "0.1020",
            "money": "0.00783483",
            "fee": "0.01708790",
            "pay_bix": 1,
            "fee_symbol": "BIX"
        },
        "id": "686384927",
        "order": null,
        "timestamp": 1528753861000,
        "datetime": "2018-06-11T17:51:01-04:00",
        "symbol": "ETH\/BTC",
        "type": "limit",
        "takerOrMaker": null,
        "side": "sell",
        "price": 0.07681212,
        "amount": 0.102,
        "cost": 0.00783483624,
        "fee": {
            "cost": 0.0170879,
            "currency": null,
            "rate": null
        }
    },

Could you please make small changes
in fetch order and fetch closed orders

 take  "cost" from "deal_money" and price from  "deal_price"

also here is some mismatch between deal_price in order and trade (and i dont where it happens in cxxt or bibox)

File: js/bibox.js (hunk 1 above)
Comment (benediamond) [reply js/bibox.js]: btw, how did you get this dict to print right-justified like this?

Comment (kroitor) [reply js/bibox.js]: @benediamond using [string.ify](https://github.com/xpl/string.ify) (https://www.npmjs.com/package/string.ify) + [ololog](https://github.com/xpl/ololog) (https://www.npmjs.com/package/ololog):

```
mbp:ccxt igorkroitor$ node examples/js/cli kraken market ETH/BTC --load-markets
kraken.market (ETH/BTC)
{     limits: { amount: { min: 0.02, max: 100000000 },
                 price: { min: 0.00001, max: undefined },
                  cost: { min: 0, max: undefined }        },
   precision: { amount: 8, price: 5 },
   tierBased:    true,
  percentage:    true,
       taker:    0.0026,
       maker:    0.0016,
       tiers: { taker: [ [0, 0.0026],
                         [50000, 0.0024],
                         [100000, 0.0022],
                         [250000, 0.002],
                         [500000, 0.0018],
                         [1000000, 0.0016],
                         [2500000, 0.0014],
                         [5000000, 0.0012],
                         [10000000, 0.0001] ],
                maker: [ [0, 0.0016],
                         [50000, 0.0014],
                         [100000, 0.0012],
                         [250000, 0.001],
                         [500000, 0.0008],
                         [1000000, 0.0006],
                         [2500000, 0.0004],
                         [5000000, 0.0002],
                         [10000000, 0]      ]  },
          id:   "XETHXXBT",
      symbol:   "ETH/BTC",
        base:   "ETH",
       quote:   "BTC",
      baseId:   "XETH",
     quoteId:   "XXBT",
    darkpool:    false,
        info: {             altname:   "ETHXBT",
                        aclass_base:   "currency",
                               base:   "XETH",
                       aclass_quote:   "currency",
                              quote:   "XXBT",
                                lot:   "unit",
                      pair_decimals:    5,
                       lot_decimals:    8,
                     lot_multiplier:    1,
                       leverage_buy: [2, 3, 4, 5],
                      leverage_sell: [2, 3, 4, 5],
                               fees: [ [0, 0.26],
                                       [50000, 0.24],
                                       [100000, 0.22],
                                       [250000, 0.2],
                                       [500000, 0.18],
                                       [1000000, 0.16],
                                       [2500000, 0.14],
                                       [5000000, 0.12],
                                       [10000000, 0.1]  ],
                         fees_maker: [ [0, 0.16],
                                       [50000, 0.14],
                                       [100000, 0.12],
                                       [250000, 0.1],
                                       [500000, 0.08],
                                       [1000000, 0.06],
                                       [2500000, 0.04],
                                       [5000000, 0.02],
                                       [10000000, 0]    ],
                fee_volume_currency:   "ZUSD",
                        margin_call:    80,
                        margin_stop:    40                 },
     altname:   "ETHXBT",
      active:    true                                         }
mbp:ccxt igorkroitor$
```

https://github.com/ccxt/ccxt/tree/master/examples/js/cli.js

**Comment by benediamond** [comment]
but @Showgofar, if the order hasn't been filled yet, then will Bibox's `orderpending/order` still contain values for `deal_money` and `deal_price`?

File: js/bibox.js (hunk 1 above)
Comment (benediamond) [reply js/bibox.js]: interesting... I'd like to find a way to output strings like this from within python.

**Comment by Showgofar** [comment]
nope , you are wrong.
This order order is filled. 
"status": "closed",

but real order cost is "0.00783483" - not 0.00783054



> 
>     also here is some mismatch between deal_price in order and trade (and i dont where it happens in cxxt or bibox)
> 
> not sure what you mean by this.

I noticed mismatch between price in order and trade
"price": 0.07681212,
"deal_price": "0.07681211",

real trade price is  0.07681212...

**Comment by kroitor** [comment]
@Showgofar if would be really helpful if you could paste your entire verbose request and response. Thx in advance!

**Comment by Showgofar** [comment]
here is fetch order  by id 658851650
$response = $ccxt[$market]->$method($id, $pair['pair_symbol']);

```
Response:

{
    "info": {
        "id": 658851650,
        "createdAt": 1529322900000,
        "account_type": 0,
        "pair": "ETH_BTC",
        "coin_symbol": "ETH",
        "currency_symbol": "BTC",
        "order_side": 1,
        "order_type": 2,
        "price": "0.07663000",
        "deal_price": "0.07659785",
        "amount": "0.2290",
        "money": "0.01754827",
        "deal_amount": "0.2290",
        "deal_percent": "100.00%",
        "deal_money": "0.01754090",
        "status": 3,
        "unexecuted": "0.0000",
        "order_from": 6
    },
    "id": "658851650",
    "timestamp": 1529322900000,
    "datetime": "2018-06-18T07:55:00-04:00",
    "lastTradeTimestamp": null,
    "symbol": null,
    "type": "limit",
    "side": "buy",
    "price": 0.07663,
    "amount": 0.229,
    "cost": 0.01754827,
    "filled": 0.229,
    "remaining": 0,
    "status": "closed",
    "fee": null
}
```

And here is fetch  my trades:

```

Response:

[
    {
        "info": {
            "id": 715210575,
            "createdAt": 1529322900000,
            "account_type": 0,
            "coin_symbol": "ETH",
            "currency_symbol": "BTC",
            "order_side": 1,
            "order_type": 2,
            "price": "0.07659786",
            "amount": "0.2290",
            "money": "0.01754090",
            "fee": "0.04667618",
            "pay_bix": 1,
            "fee_symbol": "BIX"
        },
        "id": "715210575",
        "order": null,
        "timestamp": 1529322900000,
        "datetime": "2018-06-18T07:55:00-04:00",
        "symbol": "ETH\/BTC",
        "type": "limit",
        "takerOrMaker": null,
        "side": "buy",
        "price": 0.07659786,
        "amount": 0.229,
        "cost": 0.01754090994,
        "fee": {
            "cost": 0.04667618,
            "currency": "BIX",
            "rate": null
        }
    }
]

```

PS в телеграме можно быстрей решить

**Pull request merged**